nixos/modules/services/misc/redmine.nix

```diff
@@ -151,43 +224,80 @@ in
         subversion
       ];
       preStart = ''
-        # start with a fresh config directory every time
-        rm -rf ${cfg.stateDir}/config
-        cp -r ${pkgs.redmine}/share/redmine/config.dist ${cfg.stateDir}/config
+        # ensure cache directory exists for db:migrate command
+        mkdir -p ${cfg.stateDir}/cache
 
-        # create the basic state directory layout pkgs.redmine expects
-        mkdir -p /run/redmine
+        # create the basic directory layout the redmine package expects
+        mkdir -p /run/redmine/public
 
         for i in config files log plugins tmp; do
           mkdir -p ${cfg.stateDir}/$i
-          ln -fs ${cfg.stateDir}/$i /run/redmine/$i
+          ln -fs ${cfg.stateDir}/$i /run/redmine/
         done
 
-        # ensure cache directory exists for db:migrate command
-        mkdir -p ${cfg.stateDir}/cache
+        for i in plugin_assets themes; do
+          mkdir -p ${cfg.stateDir}/public/$i
+          ln -fs ${cfg.stateDir}/public/$i /run/redmine/public/
+        done
+
+
+        # start with a fresh config directory
+        # the config directory is copied instead of linked as some mutable data is stored in there
+        rm -rf ${cfg.stateDir}/config/*
```

Comment:
Now that I am reading a change in the Redmine launch scripts,  I think that `${cfg.stateDir}` should maybe be single-quoted (or fully escaped for shell?)

`rm -rf` as root just sounds a bit too risky with a possibly-space-containing filename…

Reply:
Very good point! I'll correct shortly. Thanks!